app/routes/index.js

```diff
@@ -151,35 +153,33 @@ router.get('/authors', async (req, res) => {
 })
 
 //User profiles route
-router.get('/authors/:id', (req, res) => {
-    if(req.params.id === undefined) return res.sendStatus(500)
-    if(!validator.isAlphanumeric(req.params.id)) return res.sendStatus(500)
-
-    User.findById(req.params.id, (err, foundUser) => {
-        if(err){
-            req.flash("error", "Oops! Something went wrong!")
-            req.log('GET Author Profile ID:', req.params.id, err)
-            return res.redirect('/')
+router.get('/authors/:link', async (req, res) => {
+    if(!req.params.link) return res.sendStatus(500)
+    const encodedLink = encodeURIComponent(req.params.link.replace(/\s/g, '-'))
```

Comment:
Maybe it would be better to just extract this combination to utils, you use it all over the place. Or maybe it's better to have middleware that does that to the link param?

Reply:
Not all links need replaced, and not all links are of article ID, it would be a messy middleware

Reply:
Also I believe that encodeURI should be done on frontend, not backend. encodeURI is for browser. On backend you can decode it at most

Reply:
I understand that you might want to place it in backend before saving to `.link` prop on mongo document, consider which option is less of a mess

Reply:
Why do we actually encode link that already should be working, you've been redirected here using it through express router after all.

Reply:
Under the philosophy of never trust user data I believe it would be better to keep the backend usage of this.

Reply:
I'm not sure I've been clear enough. `encodeURIComponent` is not a security layer, it's for ensuring that the value is a valid string and that express' router will catch it